nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/layers/convolution/MaxPoolWithArgmax.java

```diff
@@ -0,0 +1,300 @@
+                .type(Pooling2D.Pooling2DType.MAX)
+                .isSameMode(isSameNode)
+                .kH(kernelShape.get(0).intValue())
+                .kW(kernelShape.size() < 2 ? kernelShape.get(0).intValue() : kernelShape.get(1).intValue())
+                .pH(padding.get(0).intValue())
+                .pW(padding.size() < 2 ? padding.get(0).intValue() : padding.get(1).intValue())
+                .build();
+        this.config = pooling2DConfig;
+        addArgs();
+    }
+
+
+    @Override
+    public Map<String, Map<String, PropertyMapping>> mappingsForFunction() {
+        Map<String, Map<String, PropertyMapping>> ret = new HashMap<>();
+        Map<String, PropertyMapping> map = new HashMap<>();
+        val strideMapping = PropertyMapping.builder()
+                .tfAttrName("strides")
+                .onnxAttrName("strides")
+                .propertyNames(new String[]{"sW", "sH"})
+                .build();
+
+        val paddingMapping = PropertyMapping.builder()
+                .onnxAttrName("padding")
+                .tfAttrName("padding")
+                .propertyNames(new String[]{"pH", "pW"})
+                .build();
+
+        val kernelMapping = PropertyMapping.builder()
+                .propertyNames(new String[]{"kH", "kW"})
+                .tfInputPosition(1)
+                .onnxAttrName("ksize")
+                .build();
+
+        val dilationMapping = PropertyMapping.builder()
+                .onnxAttrName("dilations")
+                .propertyNames(new String[]{"dW", "dH"})
+                .tfAttrName("rates")
+                .build();
+
+
+        //data_format
+        val dataFormatMapping = PropertyMapping.builder()
+                .propertyNames(new String[]{"isNHWC"})
+                .tfAttrName("data_format")
+                .build();
+
+        map.put("sW", strideMapping);
+        map.put("sH", strideMapping);
+        map.put("kH", kernelMapping);
+        map.put("kW", kernelMapping);
+        map.put("dW", dilationMapping);
+        map.put("dH", dilationMapping);
+        map.put("pH", paddingMapping);
+        map.put("pW", paddingMapping);
+        map.put("isNHWC", dataFormatMapping);
+
+        ret.put(onnxName(), map);
+        ret.put(tensorflowName(), map);
+        return ret;
+    }
+
+    @Override
+    public String opName() {
+        return "max_pool_with_argmax";
+    }
+
+    @Override
+    public String onnxName() {
+        return "MaxPoolWithArgmax";
+    }
+
+    @Override
+    public String tensorflowName() {
+        return "MaxPoolWithArgmax";
+    }
+
+    @Override
+    public List<DataType> calculateOutputDataTypes(List<DataType> inputDataTypes){
+        Preconditions.checkState(inputDataTypes != null && inputDataTypes.size() == 1, "Expected 1 input data type for %s, got %s", getClass(), inputDataTypes);
+        List<DataType> result = new ArrayList<>();
+        result.add(inputDataTypes.get(0));
+        result.add(DataType.LONG);
```

Comment:
TF also allows output datatype to be configured - INT32 or INT64

Reply:
Done